### PR TITLE
Update receipt table and receipt form

### PR DESCRIPTION
**Pull request by Zack-Ren**
Sorry for the massive PR lol
Some of the interactions are not fully worked out but I added some mock data so we know what it could look like. 

Hopefully the following screenshots help in visualizing what all the code changes are.

Updated receipt table:
image

On press of `Total`:
image

UI change of receipt form:
image

Edit history when clicking on the icon in table:
image

### CODE REVIEW

File: web/components/receiptForm.tsx [hunk 1]
```diff
@@ -31,11 +32,69 @@ export const ReceiptForm = () => {
...
+                <Input required type="date" className="max-w-52" />
+              </div>
+              <div className="flex items-center justify-between">
+                <div className="flex items-center">
+                  <label className="text-sm">Total</label>
+                  <Tooltip content="The sum of all items in this expense">
+                    <CircleHelp className="w-4 h-4 ml-2" />
+                  </Tooltip>
+                </div>
+                <Input
```
Comment (sunb26) [inline web/components/receiptForm.tsx:90]: Should we indicate to the user that the total is auto-calculated and can't be edited? Either changing the shade of the box or something along those lines?

File: web/components/receiptForm.tsx [hunk 2]
```diff
@@ -31,11 +32,69 @@ export const ReceiptForm = () => {
...
+                  type="number"
+                  startContent="$"
+                  placeholder="0.00"
+                  className="max-w-52"
+                />
+              </div>
+              <div className="flex items-center justify-between">
+                <div className="flex items-center">
+                  <label className="text-sm">Gratuity</label>
+                  <Tooltip content="The gratuity provided on this expense">
```
Comment (sunb26) [inline web/components/receiptForm.tsx:68]:
```suggestion
                  <Tooltip content="The gratuity listed on this receipt.">
```

Comment (sunb26) [reply web/components/receiptForm.tsx:68]: Might be good to keep our language consistent. I've been defining an expense as an individual item on a receipt

File: web/components/receiptForm.tsx [hunk 3]
```diff
@@ -31,11 +32,69 @@ export const ReceiptForm = () => {
...
-              <Input required type="number" label="GST/HST" startContent="$" />
-              <Input type="number" label="Gratuity" startContent="$" />
-              <Input required type="date" label="Date" />
-              <Input disabled type="number" label="Total" />
+              <div className="flex items-center justify-between">
+                <div className="flex items-center">
+                  <label className="text-sm">Subtotal</label>
+                  <Tooltip content="The pre-tax ammount">
+                    <CircleHelp className="w-4 h-4 ml-2" />
+                  </Tooltip>
```
Comment (sunb26) [inline web/components/receiptForm.tsx:40]: Love the tooltips!

File: web/components/receiptForm.tsx [hunk 4]
```diff
@@ -31,11 +32,69 @@ export const ReceiptForm = () => {
...
             <form className="flex flex-col gap-4">
-              <Input required type="number" label="Subtotal" startContent="$" />
-              <Input required type="number" label="GST/HST" startContent="$" />
-              <Input type="number" label="Gratuity" startContent="$" />
-              <Input required type="date" label="Date" />
-              <Input disabled type="number" label="Total" />
+              <div className="flex items-center justify-between">
+                <div className="flex items-center">
+                  <label className="text-sm">Subtotal</label>
+                  <Tooltip content="The pre-tax ammount">
```
Comment (sunb26) [inline web/components/receiptForm.tsx:38]:
```suggestion
                  <Tooltip content="The pre-tax amount.">
```

File: web/components/receiptForm.tsx [hunk 5]
```diff
@@ -31,11 +32,69 @@ export const ReceiptForm = () => {
...
+                  type="number"
+                  startContent="$"
+                  placeholder="0.00"
+                  className="max-w-52"
+                />
+              </div>
+              <div className="flex items-center justify-between">
+                <div className="flex items-center">
+                  <label className="text-sm">GST/HST</label>
+                  <Tooltip content="Goods and services tax/harmonized sales tax">
```
Comment (sunb26) [inline web/components/receiptForm.tsx:53]:
```suggestion
                  <Tooltip content="Goods and Services Tax/Harmonized Sales Tax">
```

File: web/components/receiptForm.tsx [hunk 6]
```diff
@@ -31,11 +32,69 @@ export const ReceiptForm = () => {
...
+                </div>
+                <Input
+                  type="number"
+                  startContent="$"
+                  placeholder="0.00"
+                  className="max-w-52"
+                />
+              </div>
+              <div className="flex items-center justify-between">
+                <label className="text-sm">Date</label>
```
Comment (sunb26) [inline web/components/receiptForm.tsx:80]: This should be the date of the receipt not the current date correct? Might be good to add a tooltip here to ensure no misinterpretations

File: web/components/receiptForm.tsx [hunk 7]
```diff
@@ -31,11 +32,69 @@ export const ReceiptForm = () => {
...
+                <Input required type="date" className="max-w-52" />
+              </div>
+              <div className="flex items-center justify-between">
+                <div className="flex items-center">
+                  <label className="text-sm">Total</label>
+                  <Tooltip content="The sum of all items in this expense">
```
Comment (sunb26) [inline web/components/receiptForm.tsx:86]:
```suggestion
                  <Tooltip content="The sum of all expenses on this receipt after tax.">
```

**Review by sunb26 — CHANGES_REQUESTED** [review]
Just some simple change suggestions. Feel free to rebuttle or ignore :)

File: web/components/receiptForm.tsx (hunk 1 above)
Comment (Zack-Ren) [reply web/components/receiptForm.tsx:90]: This field is already disabled and users cant input anything to it. See `isReadOnly` on line 91.

**Review by Lev1ty — APPROVED** [review]

File: web/components/receiptForm.tsx (hunk 1 above)
Comment (sunb26) [reply web/components/receiptForm.tsx:95]: I meant more like a visual indicator so the user doesn't think it should be editable

Comment (Zack-Ren) [reply web/components/receiptForm.tsx:95]: Maybe including something in the tooltip would be enough? Or do you think a change to the actual textbox would be better.

File: web/components/receiptForm.tsx [hunk 8]
```diff
@@ -31,11 +32,74 @@ export const ReceiptForm = () => {
...
             <form className="flex flex-col gap-4">
-              <Input required type="number" label="Subtotal" startContent="$" />
-              <Input required type="number" label="GST/HST" startContent="$" />
-              <Input type="number" label="Gratuity" startContent="$" />
-              <Input required type="date" label="Date" />
-              <Input disabled type="number" label="Total" />
+              <div className="flex items-center justify-between">
+                <div className="flex items-center">
+                  <label className="text-sm">Subtotal</label>
+                  <Tooltip content="The pre-tax ammount.">
```
Comment (sunb26) [inline web/components/receiptForm.tsx:38]:
```suggestion
                  <Tooltip content="The pre-tax amount.">
```

typo

**Review by sunb26 — APPROVED** [review]

**Pull request merged**